sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Processor/PartitionLoadBalancer.cs

```diff
@@ -0,0 +1,443 @@
+        /// </summary>
+        ///
+        public readonly string ConsumerGroup;
+
+        /// <summary>
+        ///   A unique name used to identify this event processor.
+        /// </summary>
+        ///
+        public readonly string Identifier;
+
+        /// <summary>
+        ///   The minimum amount of time to be elapsed between two load balancing verifications.
+        /// </summary>
+        ///
+        public virtual TimeSpan LoadBalanceUpdate => TimeSpan.FromSeconds(10);
+
+        /// <summary>
+        ///   The minimum amount of time for an ownership to be considered expired without further updates.
+        /// </summary>
+        ///
+        private TimeSpan OwnershipExpiration;
+
+        public IEnumerable<string> OwnedPartitionIds => InstanceOwnership.Keys;
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="PartitionLoadBalancer"/> class.
+        /// </summary>
+        ///
+        public PartitionLoadBalancer(PartitionManager storageManager,
+                                      string identifier,
+                                      string consumerGroup,
+                                      string fullyQualifiedNamespace,
+                                      string eventHubName,
+                                      TimeSpan ownershipExpiration)
+        {
+            Argument.AssertNotNull(storageManager, nameof(storageManager));
+            Argument.AssertNotNullOrEmpty(identifier, nameof(identifier));
+            Argument.AssertNotNullOrEmpty(consumerGroup, nameof(consumerGroup));
+            Argument.AssertNotNullOrEmpty(fullyQualifiedNamespace, nameof(fullyQualifiedNamespace));
+            Argument.AssertNotNullOrEmpty(eventHubName, nameof(eventHubName));
+
+            StorageManager = storageManager;
+            Identifier = identifier;
+            FullyQualifiedNamespace = fullyQualifiedNamespace;
+            EventHubName = eventHubName;
+            ConsumerGroup = consumerGroup;
+            OwnershipExpiration = ownershipExpiration;
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="PartitionLoadBalancer"/> class.
+        /// </summary>
+        ///
+        protected PartitionLoadBalancer()
+        {
+        }
+
+        /// <summary>
+        ///   Performs load balancing between multiple EventProcessorClient instances, claiming others' partitions to enforce
+        ///   a more equal distribution when necessary.  It also manages its own partition processing tasks and ownership.
+        /// </summary>
+        ///
+        /// <param name="partitionIds"></param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+        ///
+        public virtual async ValueTask<PartitionOwnership> RunLoadbalancingAsync(string[] partitionIds, CancellationToken cancellationToken)
```

Comment:
nit: If you'd be so kind, please break method parameters onto their own line for consistency.